k8s/oidc-vault/README.md

```diff
@@ -0,0 +1,458 @@
+
+   ```console
+   $ vault write auth/jwt/config oidc_discovery_url=https://oidc-discovery.example.org default_role=“dev”
+   ```
+
+3. Define a policy `my-dev-policy` that will be assigned to a `dev` role that we'll create in the next step:
+
+   Create a policy file called `vault-policy.hcl` with the following content (you can find the file [here](./vault/vault-policy.hcl)):
+
+   ```console
+   path "secret/my-super-secret" {
+      capabilities = ["read"]
+   }
+   ```
+
+   and create the policy:
+
+   ```console
+   $ vault policy write my-dev-policy vault-policy.hcl
+   ```
+
+3. Create a role `dev`, binding the subject and audience that will be in the JWT, and configuring the `sub` claim that will be used to identify the user. Also set up a 24 hour TTL for testing purposes and a policy `my-dev-policy` that will be assigned to the tokens:
+
+   ```console
+   $ vault write auth/jwt/role/dev role_type=jwt user_claim=sub bound_audiences=TESTING bound_subject=spiffe://example.org/ns/default/sa/default token_ttl=24h token_policies=my-dev-policy
+   ```
+
+## Get Vault Credentials
+
+Now we are going to get an access token to use with Vault. We'll use a sample client workload to get an identity using the SPIRE Federation feature.
+
+### Get the JWT-SVID
+
+   1. First, let's get the client pod name we created in the Kubernetes Getting Started Guide:
+
+      ```console
+      $ kubectl get pods
+      ```
+      output:
+      ```
+      NAME                      READY   STATUS    RESTARTS   AGE
+      client-7c94755d97-mq8dl   1/1     Running   1          10d
+      ```
+
+   2. Get the JWT-SVID that identifies our client workload:
+
+      ```console
+      kubectl exec client-7c94755d97-mq8dl -- /opt/spire/bin/spire-agent api fetch jwt \
+         -audience TESTING \
+         -socketPath /run/spire/sockets/agent.sock
+      ```
+
+   3. Copy the JWT from the response into your clipboard. You'll find the JWT under `token(spiffe://example.org/ns/default/sa/default)`. It looks something like this:
+
+      ```console
+      eyJhbGciOiJSUzI1NiIsImtpZCI6IjQ0c0R2cW9kRHRUUmVqR1pTMmZ4c2RUdTNuc3FmTzl6IiwidHlwIjoiSldUIn0.eyJhdWQiOlsiVEVTVElORyJdLCJleHAiOjE1ODgwOTIyODgsImlhdCI6MTU4ODA5MTk4OCwiaXNzIjoiaHR0cHM6Ly9zcGlyZS12YXVsdC1vaWRjLnNraXRhbGVlLm9yZyIsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL2RlZmF1bHQvc2EvZGVmYXVsdCJ9.Me8U9qE6yyd5mezSiMcPgwoJm2ihQZXTL-0ClAJyssg9yhCx1D4Gea3_n4pFjp86RfLiUSsGzyjBL4r0FRA6_0grJFnLdret2ynni6zZyYw6s0k38vsJIZ4rZNfY09IanQ1Ak_GW1yHVOtzRqd3vr8GgrtXzHzsWfl5YgzhWozJUYVIj1eN91aftJ-Iuvo2KYcxu1QgrIhP8Ec_6m2Kg06oRsKCb0a6C4J78wW-lXd5orDvrO2wAksmUjBwtxFA6EggtVVSKE85EG7gUgPT1xU7B2rggXC1RKUgxXqpFWHk-7qbFdk7enurxsSSGqvVSIW7KK0sYTcw5GeKze0iggQ
+      ```
+
+### Authenticate to Vault Server
+   1. Create a file called `payload.json` with your JWT from the previous step:
+
+      ```console
+      {"role": "dev","jwt": "<PASTE_YOUR_JWT_TOKEN_HERE>"}
+      ```
+
+   2. Authenticate against the Vault server using the payload:
```

Comment:
Would it be more specific to say "Vault REST API" instead of "Vault server"? Or maybe "Vault server REST API"?(Same question for my comment in part 4.) If it's unnecessary, we can leave it as is.